gravitee-node-notifier/src/main/java/io/gravitee/node/notifier/NotifierServiceImpl.java

```diff
@@ -75,19 +82,76 @@ public void register(
       );
       final String id = buildNotificationId(
         definition.getResourceId(),
+        definition.getResourceType(),
         definition.getType(),
         definition.getAudienceId()
       );
       triggers.put(id, notificationTrigger);
+      preserveNotificationDefinition(definition);
+
       notificationTrigger.start();
     } else {
       logger.debug("Node notifier service disabled");
     }
   }
 
+  private void preserveNotificationDefinition(
+    NotificationDefinition definition
+  ) {
+    final String resourceKey = buildResourceKey(
+      definition.getResourceId(),
+      definition.getResourceType()
+    );
+    List<NotificationDefinition> allDefinitions = definitions.get(resourceKey);
+    if (Objects.isNull(allDefinitions)) {
+      allDefinitions =
+        definitions.computeIfAbsent(
+          resourceKey,
+          key -> Collections.synchronizedList(new ArrayList<>())
+        );
+    }
+    allDefinitions.add(definition);
+  }
```

Comment:
I think the problem remains the same no ?
What about this:

```java
definitions.computeIfAbsent(
          resourceKey,
          key -> Collections.synchronizedList(new ArrayList<>())
        ).add(definition);
```

Reply:
:face_exhaling: :+1: